### PR TITLE
PIN-4542 Add update of lifespan and daily calls on descriptors

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4541](https://github.com/pagopa/interop-be-catalog-process/pull/236) to compile

[PIN-4541]: https://pagopa.atlassian.net/browse/PIN-4541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -2561,6 +2561,115 @@ paths:
...
+  /eservices/{eServiceId}/descriptors/{descriptorId}/update:
+    parameters:
+      - $ref: '#/components/parameters/CorrelationIdHeader'
+    post:
+      security:
+        - bearerAuth: []
+      tags:
+        - eservices
+      summary: Publish the selected descriptor.
+      operationId: updatePublishedDescriptor
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
      operationId: updateDescriptor
```

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -2561,6 +2561,115 @@ paths:
...
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        description: Payload
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateEServicePublishedDescriptorSeed'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
              $ref: '#/components/schemas/UpdateEServiceDescriptorSeed'
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -2561,6 +2561,115 @@ paths:
...
+      responses:
+        '200':
+          description: EService with published descriptor updated.
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
          description: EService descriptor updated.
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 4]
```diff
@@ -146,6 +146,34 @@ final case class EServicesApiServiceImpl(
...
+      descriptorUuid <- descriptorId.toFutureUUID
+      descriptor     <- catalogProcessService.updatePublishedDescriptor(eServiceUuid, descriptorUuid, seed.toProcess)(
+        contexts
+      )
+    } yield descriptor.toApi
+
+    onComplete(result) {
+      val headers: List[HttpHeader] = headersFromContext()
+      handleError(
+        s"Error updating published descriptor $descriptorId on service $eServiceId with seed: $seed",
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]:
```suggestion
        s"Error updating descriptor $descriptorId on service $eServiceId with seed: $seed",
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala [hunk 5]
```diff
@@ -159,6 +159,25 @@ class CatalogProcessServiceImpl(catalogProcessUrl: String, blockingEc: Execution
...
+      val request: ApiRequest[EService] =
+        api.updatePublishedDescriptor(
+          xCorrelationId = correlationId,
+          eServiceId = eServiceId,
+          descriptorId = descriptorId,
+          updateEServicePublishedDescriptorSeed = updateEServicePublishedDescriptorSeed
+        )(BearerToken(bearerToken))
+      invoker.invoke(
+        request,
+        s"Update published descriptor ${descriptorId.toString} for EService ${eServiceId.toString} with seed $updateEServicePublishedDescriptorSeed"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala]:
```suggestion
        s"Update descriptor ${descriptorId.toString} for EService ${eServiceId.toString} with seed $updateEServiceDescriptorSeed"
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala [hunk 6]
```diff
@@ -159,6 +159,25 @@ class CatalogProcessServiceImpl(catalogProcessUrl: String, blockingEc: Execution
     )
   }
 
+  override def updatePublishedDescriptor(
+    eServiceId: UUID,
+    descriptorId: UUID,
+    updateEServicePublishedDescriptorSeed: UpdateEServicePublishedDescriptorSeed
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala]:
```suggestion
    updateEServiceDescriptorSeed: UpdateEServicePublishedDescriptorSeed
```

**Review by galales — COMMENTED** [review]
LGTM, same renaming required

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**